### PR TITLE
Modify generate_container_user for all S2I images

### PR DESCRIPTION
**Pull request by tmckayus**
This change moves generate_container_user to the common directory
and references it in the Dockerfiles for each S2I image. Previously,
pyspark inherited the script from the base image while java and scala
carried separate local versions.

This change also adds the current user to /opt/app-root/etc/passwd
if the user does not exist in /etc/passwd already. The test here is
different than the original -- formerly the user was added only if
it was not 1001 or 0.

### CODE REVIEW

**Comment by tmckayus** [comment]
This is a fix for https://github.com/radanalyticsio/oshinko-s2i/issues/94

**Comment by tmckayus** [comment]
@mattf, ptal

File: common/generate_container_user [hunk 1]
```diff
@@ -2,7 +2,8 @@
 USER_ID=$(id -u)
 GROUP_ID=$(id -g)
 
-if [ x"$USER_ID" != x"0" -a x"$USER_ID" != x"1001" ]; then
+grep -e '^[^:]*:[^:]*:'$USER_ID':' /etc/passwd
+if [ "$?" != 0 ]; then
```
Comment (mattf) [inline common/generate_container_user]: would 'if [ getent passwd $USER_ID ]' work?

Comment (tmckayus) [inline common/generate_container_user]: very well might, I'll try it.
That's the problem with bash, so many ways to do things. :)

**Pull request merged**